trytond/admin.py

```diff
@@ -47,10 +69,33 @@ def run(options):
             lang.add(main_lang)
         else:
             lang = set()
+
         lang |= set(options.languages)
         pool = Pool(db_name)
-        pool.init(update=options.update, lang=list(lang),
-            activatedeps=options.activatedeps)
+
+        # Do the update only when it is needed
+        with Transaction().start(db_name, 0) as transaction:
+            cursor = transaction.connection.cursor()
+            # Lock table to upgrade
```

Comment:
Ajoute:
`# This lock will blocks others workers / processes until the current upgrade is finished`